In [1]:
import pandas as pd
df= pd.read_csv("C:\\Users\\abc\\Downloads\\PRCP-1000-ProtugeseBank\\Data\\bank-full.csv",sep=";")

# Domain Analysis

In [4]:
df.shape

(45211, 17)

Number of Instances: 45211 
Number of Attributes: 16 + output attribute. 


In [5]:
df.head()

age           job  marital  education default  balance housing loan  \
0   58    management  married   tertiary      no     2143     yes   no   
1   44    technician   single  secondary      no       29     yes   no   
2   33  entrepreneur  married  secondary      no        2     yes  yes   
3   47   blue-collar  married    unknown      no     1506     yes   no   
4   33       unknown   single    unknown      no        1      no   no   

   contact  day month  duration  campaign  pdays  previous poutcome   y  
0  unknown    5   may       261         1     -1         0  unknown  no  
1  unknown    5   may       151         1     -1         0  unknown  no  
2  unknown    5   may        76         1     -1         0  unknown  no  
3  unknown    5   may        92         1     -1         0  unknown  no  
4  unknown    5   may       198         1     -1         0  unknown  no

In [8]:
df.columns

Index(['age', 'job', 'marital', 'education', 'default', 'balance', 'housing',
       'loan', 'contact', 'day', 'month', 'duration', 'campaign', 'pdays',
       'previous', 'poutcome', 'y'],
      dtype='object')

In [12]:
print(df.job.describe())
print('-'*20)
print(df['job'].value_counts())

count           45211
unique             12
top       blue-collar
freq             9732
Name: job, dtype: object
--------------------
blue-collar      9732
management       9458
technician       7597
admin.           5171
services         4154
retired          2264
self-employed    1579
entrepreneur     1487
unemployed       1303
housemaid        1240
student           938
unknown           288
Name: job, dtype: int64


In [13]:
print(df.marital.describe())
print('-'*20)
print(df['marital'].value_counts())

count       45211
unique          3
top       married
freq        27214
Name: marital, dtype: object
--------------------
married     27214
single      12790
divorced     5207
Name: marital, dtype: int64


In [14]:
print(df.education.describe())
print('-'*20)
print(df['education'].value_counts())

count         45211
unique            4
top       secondary
freq          23202
Name: education, dtype: object
--------------------
secondary    23202
tertiary     13301
primary       6851
unknown       1857
Name: education, dtype: int64


In [15]:
print(df.default.describe())
print('-'*20)
print(df['default'].value_counts())

count     45211
unique        2
top          no
freq      44396
Name: default, dtype: object
--------------------
no     44396
yes      815
Name: default, dtype: int64


In [16]:
print(df.housing.describe())
print('-'*20)
print(df['housing'].value_counts())

count     45211
unique        2
top         yes
freq      25130
Name: housing, dtype: object
--------------------
yes    25130
no     20081
Name: housing, dtype: int64


In [17]:
print(df.loan.describe())
print('-'*20)
print(df['loan'].value_counts())

count     45211
unique        2
top          no
freq      37967
Name: loan, dtype: object
--------------------
no     37967
yes     7244
Name: loan, dtype: int64


In [18]:
print(df.contact.describe())
print('-'*20)
print(df['contact'].value_counts())

count        45211
unique           3
top       cellular
freq         29285
Name: contact, dtype: object
--------------------
cellular     29285
unknown      13020
telephone     2906
Name: contact, dtype: int64


In [19]:
print(df.month.describe())
print('-'*20)
print(df['month'].value_counts())

count     45211
unique       12
top         may
freq      13766
Name: month, dtype: object
--------------------
may    13766
jul     6895
aug     6247
jun     5341
nov     3970
apr     2932
feb     2649
jan     1403
oct      738
sep      579
mar      477
dec      214
Name: month, dtype: int64


In [20]:
print(df.poutcome.describe())
print('-'*20)
print(df['poutcome'].value_counts())

count       45211
unique          4
top       unknown
freq        36959
Name: poutcome, dtype: object
--------------------
unknown    36959
failure     4901
other       1840
success     1511
Name: poutcome, dtype: int64


In [21]:
print(df.y.describe())
print('-'*20)
print(df['y'].value_counts())

count     45211
unique        2
top          no
freq      39922
Name: y, dtype: object
--------------------
no     39922
yes     5289
Name: y, dtype: int64


In [22]:
df.describe()

age        balance           day      duration      campaign  \
count  45211.000000   45211.000000  45211.000000  45211.000000  45211.000000   
mean      40.936210    1362.272058     15.806419    258.163080      2.763841   
std       10.618762    3044.765829      8.322476    257.527812      3.098021   
min       18.000000   -8019.000000      1.000000      0.000000      1.000000   
25%       33.000000      72.000000      8.000000    103.000000      1.000000   
50%       39.000000     448.000000     16.000000    180.000000      2.000000   
75%       48.000000    1428.000000     21.000000    319.000000      3.000000   
max       95.000000  102127.000000     31.000000   4918.000000     63.000000   

              pdays      previous  
count  45211.000000  45211.000000  
mean      40.197828      0.580323  
std      100.128746      2.303441  
min       -1.000000      0.000000  
25%       -1.000000      0.000000  
50%       -1.000000      0.000000  
75%       -1.000000      0.000000  
max      871.000000    275.000000

Attribute information: 

1 - age (numeric) 

2 - job : type of job (categorical: admin.", "unknown", "unemployed", "management", "housemaid", “entrepreneur", "student", "blue-collar", "self-employed",“ retired", "technician", "services")

3 - marital : marital status (categorical: "married",“ divorced", "single"; note: "divorced" means divorced or widowed)

4 - education (categorical: "unknown", "secondary", "primary", "tertiary")

5 - default: has credit in default? (binary: "yes", "no") 

6 - balance: average yearly balance, in euros (numeric)

7 - housing: has housing loan? (binary: "yes", "no")

8 - loan: has personal loan? (binary: "yes","no")

9 - contact: contact communication type (categorical: "unknown","telephone","cellular") 

10 - day: last contact day of the month (numeric)

11 - month: last contact month of year (categorical: "jan", "feb", "mar", ..., "nov", "dec")

12 - duration: last contact duration, in seconds (numeric) # other attributes:

13 - campaign: number of contacts performed during this campaign and for this client (numeric, includes last contact)

14 - pdays: number of days that passed by after the client was last contacted from a previous campaign (numeric, -1 means client 
was not previously contacted) 

15 - previous: number of contacts performed before this campaign and for this client (numeric)

16 - poutcome: outcome of the previous marketing campaign (categorical: "unknown","other","failure","success")

Output variable: 17 - y - has the client subscribed a term deposit? (binary: "yes", "no")

-----------------------------------------------------------------------

Numeric features description:

Age- max age is 95 and min is 18 , average age is 40

balance- 

duration - 

compaign - 

pday - 

previous - 